### PR TITLE
rest client for MBMS salesforce api

### PR DESCRIPTION
**Pull request by alextanton**

### CODE REVIEW

File: app/services/time_of_need_service.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+require 'restforce'
+require 'faraday'
+
+class TimeOfNeedService
+
+  @conn = Faraday.new(:url => 'https://va--mbmssit.cs33.my.salesforce.com/services/oauth2/token')
```
Comment (goodcodeguy) [inline app/services/time_of_need_service.rb]: We need to make sure we are not hardcoding urls, please use the Settings pattern already in place by the vets-api

File: app/services/time_of_need_service.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  @request = @conn.post do |req|
+    req.headers['Content-Type'] = 'application/x-www-form-urlencoded'
+    req.body = 'client_secret=6150175763295809569&client_id=3MVG9Zdl7Yn6QDKPHMwHadV4iFddvfNHNDYs18l_CHPw5JkdgIjR0trqqefjrP4jbfu5AHnCbxyEg60Tz3KWW&api_version=42.0&username=mbms_integration@bah.com.mbmssit&password=littlekyle2spCL3xcR7dbcUrrSwcgoUrDQ&grant_type=password'
```
Comment (goodcodeguy) [inline app/services/time_of_need_service.rb]: Client Secret should NOT be stored here, please follow patterns put in place already

File: app/services/time_of_need_service.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  @client = Restforce.new(oauth_token: JSON.parse(@request.body)["access_token"],
+                         instance_url: 'https://va--MBMSSit.cs33.my.salesforce.com',
```
Comment (goodcodeguy) [inline app/services/time_of_need_service.rb]: Same as above comments, no hardcoding of URLS please.

File: config/settings.yml [hunk 4]
```diff
@@ -321,3 +321,15 @@ search:
   access_key: SEARCH_GOV_ACCESS_KEY
   affiliate: va
   url: https://search.usa.gov/api/v2
+
+# Time of need
+timeOfNneed:
```
Comment (goodcodeguy) [inline config/settings.yml]: this should be `time_of_need`. This file uses snake case and this is misspelled.

File: config/settings.yml [hunk 5]
```diff
@@ -321,3 +321,15 @@ search:
...
+  api_version: 42.0
+  username: mbms_integration@bah.com.mbmssit
+  password: littlekyle2spCL3xcR7dbcUrrSwcgoUrDQ
+  grant_type: password
+  client_secret: 6150175763295809569&client_id=3MVG9Zdl7Yn6QDKPHMwHadV4iFddvfNHNDYs18l_CHPw5JkdgIjR0trqqefjrP4jbfu5AHnCbxyEg60Tz3KWW&
```
Comment (goodcodeguy) [inline config/settings.yml]: we shouldn't be storing passwords or secrets in the settings file, please review https://department-of-veterans-affairs.github.io/va-digital-services-platform-docs/docs/vets-developer-docs/vets-api/settings.html for how we should be handling secrets.

Comment (alextanton) [reply config/settings.yml]: Are the secret keys already in the file not actual secret keys?

File: app/services/time_of_need_service.rb (hunk 3 above)
Comment (William-Dahlheim) [reply app/services/time_of_need_service.rb]: @siriusmintllc We request this branch or specifically this PR deleted as it contains sensitive data.  The data is out dated but still needs cleaned.

**Pull request merged**